overlord/hookstate/hookmgr.go

```diff
@@ -89,9 +90,14 @@ func Manager(s *state.State) (*HookManager, error) {
 }
 
 // HookTask returns a task that will run the specified hook.
-func HookTask(s *state.State, taskSummary, snapName string, revision snap.Revision, hookName string) *state.Task {
+func HookTask(s *state.State, taskSummary, snapName string, revision snap.Revision, hookName string, payload interface{}) *state.Task {
```

Comment:
Alright, done. Instead of a generic payload, it's just a `map[string]interface{}` which simply results in an initial context for the hook handler.